src/backend/distributed/executor/insert_select_executor.c

```diff
@@ -210,8 +210,31 @@ CoordinatorInsertSelectExecScanInternal(CustomScanState *node)
 							 distSelectJob->jobId);
 			char *distResultPrefix = distResultPrefixString->data;
 
-			CitusTableCacheEntry *targetRelation =
+			CitusTableCacheEntry *cachedTargetRelation =
 				GetCitusTableCacheEntry(targetRelationId);
+			CitusTableCacheEntry *targetRelation = palloc(sizeof(CitusTableCacheEntry));
+			*targetRelation = *cachedTargetRelation;
+
+#if PG_USE_ASSERT_CHECKING
```

Comment:
What is this? add comment.

Why are we setting these to null after copying?

Reply:
I set it to `NULL` to make sure it errors if used